### PR TITLE
fix chain.get_block_txn_infos getObjectArray error.

### PR DESCRIPTION
**Pull request by nkysg**
fix chain.get_block_txn_infos getObjectArray error. need revert https://github.com/starcoinorg/starcoin-java/pull/42

## Summary by CodeRabbit

- **New Features**
  - Upgraded JSON library to use `com.alibaba.fastjson` for improved performance and flexibility.
  - Added a new dependency for `jsonrpc2-client` to enhance JSON-RPC client capabilities.

- **Bug Fixes**
  - Corrected method name `getSTCTreasurBalance` to `getSTCTreasuryBalance` for consistency.

- **Refactor**
  - Updated method and import statements to align with the new JSON library.
  - Removed redundant `@JsonProperty` annotations to streamline codebase.

- **Chores**
  - Updated Maven dependency versions for better compatibility and stability.

### CODE REVIEW

**Comment by nkysg** [comment]
Most code diff see this url https://github.com/starcoinorg/starcoin-java/compare/fix_json_rpc_parse..fix_stc_supply

File: src/main/java/org/starcoin/bean/TokenInfo.java [hunk 1]
```diff
@@ -23,23 +23,19 @@
...
-    @JsonProperty("scaling_factor")
     private long scalingFactor;
 
     @JSONField(name = "burn_events")
-    @JsonProperty("burn_events")
     private Object burnEvents;
 
     @JSONField(name = "mint_events")
-    @JsonProperty("mint_events")
+
```
[the remark this answers is not in the document]
Comment (nkysg) [reply src/main/java/org/starcoin/bean/TokenInfo.java:38]: fixed

File: src/main/java/org/starcoin/api/TokenContractRPCClient.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ public BigInteger getSTCMarketCap() throws JSONRPC2SessionException {
     /**
      * 用于获取 STC 在国库中的数量
      */
-    public BigInteger getSTCTreasurBalance() throws JSONRPC2SessionException {
+    public BigInteger getSTCTreasuryBalance() throws JSONRPC2SessionException {
```
[the remark this answers is not in the document]
Comment (nkysg) [reply src/main/java/org/starcoin/api/TokenContractRPCClient.java:109]: fixed

[the remark this answers is not in the document]
Comment (nkysg) [reply src/main/java/org/starcoin/api/TokenContractRPCClient.java:109]: fixed.

**Review by welbon — APPROVED** [review]

**Pull request merged**